### PR TITLE
Setup CircleCI and automatic releases

### PR DESCRIPTION
**Pull request by arichiardi**
This patch introduce the configuration necessary for building and deploying
perun to Clojars. This happens when a tag is pushed to the master branch.

The code is also now using `degree9/boot-semver` and its `version.properties` file.

~I am just waiting for a resolution for https://github.com/degree9/boot-semver/pull/76 and then it should be good to go.~

I tried it against my custom https://clojars.org/powerlaces/perun and it worked like a charm with my own deploy tokens.

Of course I can help with the setup once this is merged - I would need admin privileges even if for a very short time for the project and also collaborator privileges on Clojars. 

Oh I also tweaked and added the [MAINTAINERS.md](https://github.com/hashobject/perun/pull/252/files#diff-39da3bd6270d44ea37b6ed50bd42eeb9d93ac5e1639645871a69cbe08cbe29de) file.

### CODE REVIEW

File: build.boot [hunk 1]
```diff
@@ -13,43 +11,29 @@
...
-       :license     {"name" "Eclipse Public License"
-                     "url"  "http://www.eclipse.org/legal/epl-v10.html"}})
-
-
-(deftask release-snapshot
-  "Release snapshot"
-  []
-  (comp (build-jar) (push-snapshot)))
+ aot {:all true}
+ pom {:project 'powerlaces/perun
```
Comment (arichiardi) [inline build.boot]: Will change this right before merging, I still need to test one thing.

**Comment by arichiardi** [comment]
This is now ready!

**Comment by arichiardi** [comment]
@allentiak @podviaznikov in order for me to complete this please add me as admin (or directly add the project to CircleCI) and as collaborator on Clojars so that I can create a deploy token...or you can add the token after I configured things but I'd rather do this myself in order to speed things up.

**Comment by allentiak** [comment]
> @allentiak @podviaznikov in order for me to complete this please add me as admin (or directly add the project to CircleCI) and as collaborator on Clojars

Done. (BTW: You were already an admin on clojars :-)

**Review by allentiak — CHANGES_REQUESTED** [review]
Great that you also updated the README!
Just remember to update the version to match your other PR.
(See comment.)

File: version.properties [hunk 2]
```diff
@@ -0,0 +1,2 @@
+#Wed Dec 16 14:41:05 PST 2020
+VERSION=0.5.0-SNAPSHOT
```
Comment (allentiak) [inline version.properties:2]: I think the current version is different to this one.
(0.5.1?)

In any case, they should be always ascendent :-)

Comment (arichiardi) [reply version.properties:2]: Yep I am going to try things out with this first, then I will definitely bump it.

**Comment by arichiardi** [comment]
Some of the tests are not working cause there are a couple of binaries missing: `pandoc`, `lessc` ... there might be more.

**Comment by arichiardi** [comment]
~As per Slack conversation, I cannot generate deploy tokens with my Clojars user.~

~I am going to merge this one in but I'd need either @allentiak or @podviaznikov to add the token to~

The :upside_down_face: is not necessary anymore. I have configured it with a token and my username so next time we push a release tag to master everything should get deployed!

**Comment by allentiak** [comment]
Thanks for your work, Andrea!

**Pull request merged**